package/boot/uboot-envtools/files/lantiq

```diff
@@ -3,32 +3,35 @@
 # Copyright (C) 2012 OpenWrt.org
 #
 
-[ -e /etc/config/ubootenv ] && exit 0
-
-touch /etc/config/ubootenv
-
 . /lib/uboot-envtools.sh
 . /lib/functions.sh
 
+idx="$(find_mtd_index u-boot-env)"
+
+[ -n "$idx" ] || exit 0
+
+[ -e /etc/config/ubootenv -a "$(uci get ubootenv.@ubootenv[-1].dev)" = "/dev/mtd$idx" ] && exit 0
```

Comment:
On a second view/thought, this is a NAK for me.

It is common and thus assumed behavior that existing uci config files do not get changed/"updated" by these config generation scripts. This is also important in the context of consistency between these config files. (E.g. if you want updated network config, you would delete /etc/config/network).

I'd say it's fine to provide additional migration scripts where this is really necessary, but I'm against mixing initial creation of uci config files and changes in the same scripts.

Reply:
By enabling `CONFIG_MTD_PARTITIONED_MASTER` the numbering of mtd partitions will be changed on all devices of this target. Adding an additional dedicated migration script would be possible, but just adds a lot of redundant code and makes the existing script de facto superfluous.

Reply:
Sorry, but that sounds like another reason against enabling CONFIG_MTD_PARTITIONED_MASTER unless there is a need for it.